WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java

```diff
@@ -621,11 +634,29 @@ public void onCommentChanged(OnCommentChanged event) {
         if (mQuickActionProcessor == null) {
             return;
         }
+
         if (event.causeOfChange == CommentAction.PUSH_COMMENT) {
             if (event.isError()) {
                 mQuickActionProcessor.requestFailed(ARG_ACTION_APPROVE);
             } else {
-                mQuickActionProcessor.requestCompleted(ARG_ACTION_APPROVE);
+                if (mActionedCommentsRemoteIds.size() > 0) {
+                    // here we need to check ids
+                    // CommentModel localComment = mCommentStore.getCommentBySiteAndRemoteId(site, remoteCommendId);
+                    int containedCount = 0;
+                    for (Integer commentLocalId : event.changedCommentsLocalIds) {
+                        CommentModel localComment = mCommentStore.getCommentByLocalId(commentLocalId);
+                        if (localComment != null) {
+                            if (mActionedCommentsRemoteIds.contains(localComment.getRemoteCommentId())) {
+                                containedCount++;
+                            }
+                        }
+                    }
+
+                    if (containedCount != mActionedCommentsRemoteIds.size()) {
```

Comment:
Not quite sure this will work fine when there are multiple quick actions ongoing in background.
What does happen when ` event.changedCommentsLocalIds` is a list that only contains the ID of the first comment being moderated?